great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -267,6 +290,49 @@ def config_variables(self) -> Dict:
             self._config_variables = self._load_config_variables()
         return self._config_variables
 
+    @property
+    def config(self) -> DataContextConfig:
+        return self._project_config
+
+    @property
+    def root_directory(self) -> Optional[str]:
+        """The root directory for configuration objects in the data context; the location in which
+        ``great_expectations.yml`` is located.
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+
+        """
+        if hasattr(self, "_context_root_directory"):
+            return self._context_root_directory
+        return
+
+    @property
+    def project_config_with_variables_substituted(self) -> DataContextConfig:
+        return self.get_config_with_variables_substituted()
+
+    @property
+    def plugins_directory(self):
+        """The directory in which custom plugin modules should be placed.
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+        """
+        return self._normalize_absolute_or_relative_path(
+            self.project_config_with_variables_substituted.plugins_directory
+        )
+
+    def _normalize_absolute_or_relative_path(
+        self, path: Optional[str]
+    ) -> Optional[str]:
+        """
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+        """
+        if path is None:
+            return
+        if os.path.isabs(path):
+            return path
+        else:
+            return os.path.join(self.root_directory, path)
```

Comment:
Nit - Minor refactor 
```suggestion
        if path is None or os.path.isabs(path):
            return path
        return os.path.join(self.root_directory, path)
```